src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -241,9 +244,26 @@ class ProtobufPlugin implements Plugin<Project> {
       sourceSetNames.each { sourceSetName ->
         def extractProtosTask = maybeAddExtractProtosTask(sourceSetName)
         generateProtoTask.dependsOn(extractProtosTask)
+      }
 
-        def extractIncludeProtosTask = maybeAddExtractIncludeProtosTask(sourceSetName)
+      // Extract proto file from the compile classpath configuration exposed in Android Gradle plugin 2.5.  Revert to
+      // original behavior if the configuration is not found.
```

Comment:
Just say the if-branch is for Android plugin 2.5+, and the else-branch is for below 2.5.
There is no "original" behavior outside the context of this PR.